src/LSST/control/ControlMain.cpp

```diff
@@ -57,16 +66,161 @@ ControlMain::Ptr ControlMain::getPtr() {
+    _simCore->start();
+    faultmgr::FaultMgr::setup();
+    control::FpgaIo::setup(_simCore);
+    control::MotionEngine::setup();
+    control::Context::setup();
+
+    // Register the PowerSystem with FpgaIo so that it gets updates.
+    auto powerSys = control::Context::get()->model.getPowerSystem();
+    control::FpgaIo::get().registerPowerSys(powerSys);
+
+    // Setup the control system
+    auto context = control::Context::get();
+    context->model.ctrlSetup();
+
+    // At this point, MotionCtrl and FpgaCtrl should be configured. Start the control loops
+    context->model.ctrlStart();
+
+    // At this point, the LabView code seems to want to put the system into
+    // ReadyIdle. We're NOT doing that. The system is going into StandbyState
+    // until it gets an explicit command to do something else.
+
+    // Start the telemetry server
+    // FUTURE: Telemetry for the GUI requires the ComControlServer->ComConnection::welcomeMsg to
+    //         work properly. Those elements should be in the telemetry so there's no need for a
+    //         ComControlServer connection for Telemetry. This has to wait until the existing
+    //         controller is no longer used.
+    LINFO("Starting Telemetry Server");
+    // FUTURE: get the correct entries into `system::Config`.
+    int const telemPort = 50001;
+    auto servTelemetryMap = system::TelemetryMap::Ptr(new system::TelemetryMap());
+    auto telemetryServ = system::TelemetryCom::create(servTelemetryMap, telemPort);
+
+    telemetryServ->startServer();
+    if (!telemetryServ->waitForServerRunning(5)) {
+        LCRITICAL("Telemetry server failed to start.");
+        exit(-1);  // change to powerdown and exit
+    }
+
+    // Wait for MotionCtrl to be ready
+    context->model.waitForCtrlReady();
+
+    // Start a ComControlServer
+    LDEBUG("ComControlServer starting...");
+    system::IoContextPtr ioContext = make_shared<boost::asio::io_context>();
+    int port = system::Config::get().getControlServerPort();
+    auto cmdFactory = control::NetCommandFactory::create();
+    system::ComControl::setupNormalFactory(cmdFactory);
+    _comServer = system::ComControlServer::create(ioContext, port, cmdFactory, true);
+    LINFO("ComControlServer created port=", port);
+
+    atomic<bool> comServerDone{false};
+    auto comServState = _comServer->getState();
+    LDEBUG("ComControlServer comServState=", _comServer->prettyState(comServState));
+
+    auto cServ = _comServer;
+    thread comControlServThrd([&cServ, &comServerDone, &log]() {
+        LINFO("server run ", cServ->prettyState(cServ->getState()));
+        log.flush();
+        cServ->run();
+        LINFO("server finish");
+        comServerDone = true;
+    });
+
+    // wait for the server to be running
+    {
+        int j = 0;
+        int const maxSeconds = 30;
+        for (; (cServ->getState() != LSST::m2cellcpp::system::ComServer::RUNNING) && j < maxSeconds; ++j) {
+            sleep(1);
```

Comment:
Why not use the modern C++11 for this?
```
std::this_thread::sleep_for(1000ms);
```
See: https://en.cppreference.com/w/cpp/thread/sleep_for